pkg/generate/git/repository.go

```diff
@@ -337,6 +348,42 @@ func command(w io.Writer, name, dir string, env []string, args ...string) (stdou
 	return
 }
 
+// timedCommand executes an external command in the given directory with a timeout.
+// The command's standard out and error are trimmed and returned as strings
+// It may return the type *GitError if the command itself fails.
+func timedCommand(timeout time.Duration, w io.Writer, name, dir string, env []string, args ...string) (stdout, stderr string, err error) {
```

Comment:
> Yes, but I wanted to make this have minimal impact on any existing consumers of the command() call. @csrwng posted in the issue that we don't want timeouts on any other call.

Passing `timeout <= 0` could be "no timeout" pretty naturally.  You've created a lot of code here that doesn't need to be.